### PR TITLE
[#477] Remove `neat-interpolation`

### LINKED ISSUE #477: Remove `neat-interpolation`
It prevents us from being on Stackage.

As the messy workaround of the breaking design of the `neat-interpolation`, we could use manual indentation and `Text` appending.

### PR DESCRIPTION
**Pull request by vrom911**
Resolves #477

I am sooo grateful to out tests! 🙏 
Thanks @chshersh ❤️

### CODE REVIEW

File: summoner-cli/src/Summoner/Default.hs [hunk 1]
```diff
@@ -75,95 +72,95 @@ currentYear = do
...
-    #     package = "relude"
-    #     module  = "Relude"
-|]
+defaultConfigFileContent = unlines
+    [ "# This file is automatically generated by Summoner."
+    , "# Edit required fields, uncomment additional settings or delete irrelevant"
+    , "# lines for the personalized configuration."
+    , ""
+    , "# GitHub user name"
+    , "owner = \"" <> defaultOwner <> "\""
```
Comment (chshersh) [inline summoner-cli/src/Summoner/Default.hs:81]: I feel like we will benefit from having the following function in `Summoner.Text`:

```haskell
quote :: Text -> Text
quote t = "\"" <> t <> "\""
```

Not necessary required right now, but probably will be useful in the future, so something to keep in mind.

File: summoner-cli/src/Summoner/Template/Cabal.hs [hunk 2]
```diff
@@ -43,7 +39,6 @@ cabalFile Settings{..} = File (toString settingsRepo ++ ".cabal") cabalFileConte
         , memptyIfFalse settingsBench $ benchmarkStanza $ memptyIfFalse settingsIsLib $ ", " <> settingsRepo
         ]
 
-    -- TODO: do something to not have empty lines
```
Comment (chshersh) [inline summoner-cli/src/Summoner/Template/Cabal.hs:46]: Something has been done :laughing:

File: summoner-cli/src/Summoner/Template/Cabal.hs [hunk 3]
```diff
@@ -80,139 +75,125 @@ cabalFile Settings{..} = File (toString settingsRepo ++ ".cabal") cabalFileConte
...
+    ghcOptions :: [Text]
     ghcOptions = case settingsGhcOptions of
-        [] -> defaultGhcOptions
-        x:xs ->
-            let customGhcOptions = T.intercalate "\n" $ x : map (T.replicate 21 " " <>) xs in
-            [text|
-            ghc-options:         $customGhcOptions
-            |]
+        []   -> defaultGhcOptions
+        x:xs -> "ghc-options:         " <> x : map (T.replicate 21 " " <>) xs
```
Comment (chshersh) [inline summoner-cli/src/Summoner/Template/Cabal.hs:105]: Maybe we can use `indent 21` from `colourista` here? Also, `indent` is implemented using `sconcat` :slightly_smiling_face:

File: summoner-cli/src/Summoner/Template/Cabal.hs [hunk 4]
```diff
@@ -80,139 +75,125 @@ cabalFile Settings{..} = File (toString settingsRepo ++ ".cabal") cabalFileConte
...
-
-          default-language:    Haskell2010
-    |] <> defaultExtensions
+    commonStanza = unlines $
+        [ ""
+        , "common common-options"
+        , "  build-depends:       base " <> baseBounds
+        ]
+        <> customPrelude
+        <> ("" : map ("  " <>) ghcOptions)
```
Comment (chshersh) [inline summoner-cli/src/Summoner/Template/Cabal.hs:92]: Can use `indent` here as well probably. Or maybe can add 2 spaces directly to `ghc-options`?
```suggestion
        <> ("" : map (indent 2) ghcOptions)
```

File: summoner-cli/src/Summoner/Template/Cabal.hs [hunk 5]
```diff
@@ -80,139 +75,125 @@ cabalFile Settings{..} = File (toString settingsRepo ++ ".cabal") cabalFileConte
...
+    defaultExtensions :: [Text]
     defaultExtensions = case settingsExtensions of
-        [] -> ""
-        xs -> "  default-extensions:  "
-           <> T.intercalate "\n                       " xs
-           <> "\n"
+        [] -> []
+        x:xs ->
+             "  default-extensions:  " <> x
+           : map ("                       " <>) xs
```
Comment (chshersh) [inline summoner-cli/src/Summoner/Template/Cabal.hs:199]: Should it be `indent 21` here as well? :thinking:

File: summoner-cli/src/Summoner/Template/GitHub.hs [hunk 6]
```diff
@@ -112,201 +108,186 @@ gitHubFiles Settings{..} = concat
...
+        map travisCabalMatrixItem settingsTestedVersions
 
     travisCabalMatrixItem :: GhcVer -> Text
-    travisCabalMatrixItem (showGhcVer -> ghcV) = [text|- ghc: $ghcV|]
+    travisCabalMatrixItem (showGhcVer -> ghcV) = "  - ghc: " <> ghcV
 
     -- Due to the Stack issues with newer Cabal versions TravisCI for 'oldGhcs'
     -- can fail. Possible failure jobs are added to the @allow-failures@ section.
-    travisStackMtr :: Text
-    travisStackMtr = memptyIfFalse settingsStack $ tconcatMap
```
Comment (chshersh) [inline summoner-cli/src/Summoner/Template/GitHub.hs:217]: In that case, do we still use `tconcatMap`?

File: summoner-cli/src/Summoner/Template/Cabal.hs (hunk 3 above)
Comment (vrom911) [reply summoner-cli/src/Summoner/Template/Cabal.hs:105]: Oh yeah, this will be so much better! 👍

**Pull request merged**